libs/sdk-core/src/breez_services.rs

```diff
@@ -768,6 +768,9 @@ impl BreezServices {
         // update node state and channels state
         self.persister.set_node_state(&new_data.node_state)?;
 
+        // Set the default LSP, if not already
+        self.set_default_lsp_id().await?;
```

Comment:
@dangeross  Did you see that the `connect_lsp_peer` is called above (line 754)?
I think it would be clean to encapsulate all the logic in that function. What do you think?

Reply:
I separated it out because on the first sync/run the node state isn't set yet during the connect_lsp_peer call

Reply:
If encapsulating it in `connect_lsp_peer` and handling that the node state is not persisted yet, it would mean having to call `connect_lsp_peer` twice, the second being after persisting the node state.